packages/common/src/ContractInteractor.ts

```diff
@@ -397,16 +401,85 @@ export class ContractInteractor {
   }
 
   async getPastEventsForHub (extraTopics: string[], options: PastEventOptions, names: EventName[] = ActiveManagerEvents): Promise<EventData[]> {
-    return await this._getPastEvents(this.relayHubInstance.contract, names, extraTopics, options)
+    return await this._getPastEventsPaginated(this.relayHubInstance.contract, names, extraTopics, options)
   }
 
   async getPastEventsForStakeManager (names: EventName[], extraTopics: string[], options: PastEventOptions): Promise<EventData[]> {
     const stakeManager = await this.stakeManagerInstance
-    return await this._getPastEvents(stakeManager.contract, names, extraTopics, options)
+    return await this._getPastEventsPaginated(stakeManager.contract, names, extraTopics, options)
   }
 
   async getPastEventsForPenalizer (names: EventName[], extraTopics: string[], options: PastEventOptions): Promise<EventData[]> {
-    return await this._getPastEvents(this.penalizerInstance.contract, names, extraTopics, options)
+    return await this._getPastEventsPaginated(this.penalizerInstance.contract, names, extraTopics, options)
+  }
+
+  async getPagesForBlockWindow (fromBlock: BlockNumber = 1, toBlock?: BlockNumber): Promise<number> {
```

Comment:
Pages *Per* Block Window?

Reply:
No, opposite of that. "Per" implies there are multiple 'windows' consumed, like "miles per gallon", "kilometers per hour" etc.
Here there is one "window" being split into pages. 
I will rename "window" to range and the method to `getLogsPagesForRange` to make it more clear.